api/views/user_views.py

```diff
@@ -223,6 +265,47 @@ def post(self, request):
         return JsonResponse({"message": "User logged out successfully"}, status=status.HTTP_200_OK)
 
 
+@extend_schema(tags=["Authentication Endpoint"])
+class JWTLogoutView(APIView):
+    serializer_class = LogoutSerializer
+    permission_classes = [IsAuthenticated]
+
+    def post(self, request):
+        logger.info(f"Received logout request from user {request.user.id}")
+
+        try:
+            refresh_token = request.data.get("refresh")
+            if not refresh_token:
+                logger.warning("Refresh token is missing in the request")
+                return Response(
+                    {"detail": "Refresh token is required"}, status=status.HTTP_400_BAD_REQUEST
+                )
+
+            token = RefreshToken(refresh_token)
+
+            # Ensure the token belongs to the authenticated user
+            if token["user_id"] != request.user.id:
+                logger.warning(
+                    f"User {request.user.id} attempted to log out with a token not belonging to them"
+                )
+                return Response(
+                    {"detail": "Invalid token for the authenticated user"},
+                    status=status.HTTP_403_FORBIDDEN,
+                )
```

Comment:
I'm more comfortable making the log type here `critical`, and let's provide a more generic detail message to the client (e.g. "An error occurred while logging out.")

We'll have to connect these log type to Slack channels soon.

Reply:
I beleive `warning` fits better here as `critical` is typically for errors that cause the application to crash or enter some other bad state. As this is a security issue and not necessarily an application-crashing issue, I think `warning` is better.

I'll address the other issue promptly!

Reply:
You're right that `critical` is used for errors; the idea was to flag this warning as urgent; we can can keep this `warning` and discuss that during one of our upcoming meetings also.